components/purchase_modal/purchase_modal.tsx

```diff
@@ -1060,10 +1101,63 @@ class PurchaseModal extends React.PureComponent<Props, State> {
                             </PaymentDetails>
                         </div>
                     )}
+                    <div className='shipping-address-section'>
+                        <input
+                            id='address-same-than-billing-address'
+                            className='Form-checkbox-input'
+                            name='terms'
+                            type='checkbox'
+                            checked={this.state.billingSameAsShipping}
+                            onChange={() =>
+                                this.handleShippingSameAsBillingChange(
+                                    !this.state.billingSameAsShipping,
+                                )
+                            }
+                        />
+                        <span className='Form-checkbox-label'>
+                            <button
+                                onClick={() =>
+                                    this.handleShippingSameAsBillingChange(
+                                        !this.state.billingSameAsShipping,
+                                    )
+                                }
+                                type='button'
+                                className='no-style'
+                            >
+                                <span className='billing_address_btn_text'>
+                                    {
+                                        'My shipping address is the same as my billing address'
+                                    }
+                                </span>
+                            </button>
+                        </span>
+                    </div>
+                    {!this.state.billingSameAsShipping && (
+                        <AddressForm
+                            onAddressChange={this.onShippingInput}
+                            onBlur={() => {}}
+                            title={{
+                                id: 'payment_form.shipping_address',
+                                defaultMessage: 'Shipping Address',
+                            }}
+                            formId={'shippingAddress'}
+
+                            // Setup the initial country based on their billing country, or USA.
+                            initialAddress={
+                                {
+                                    country:
```

Comment:
Could we make a helper to create a full blank address? There's some issues with the type coercion:
* the underlying inputs that receive an `undefined` at first will change from being "uncontrolled" to "controlled" inputs as they get filled in. This goes against what React wants and creates console errors.
* It can be hard to ensure all the places that trust and assume the fields match the types won't explode or have unexpected behavior. AFAICT it works out at the moment, but its another place that lends itself to hard to detect regressions.

Reply:
Added a helper. I always forget Typescript isn't like Go where it will create an entire struct including all fields with zero values